app/src/main/java/fr/free/nrw/commons/category/CategoryImageUtils.java

```diff
@@ -0,0 +1,146 @@
+package fr.free.nrw.commons.category;
+
+import android.content.Context;
+
+import org.jsoup.Jsoup;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import fr.free.nrw.commons.Media;
+import timber.log.Timber;
+
+public class CategoryImageUtils {
+
+    public static List<Media> getMediaList(Context context, NodeList childNodes) {
+        List<Media> categoryImages = new ArrayList<>();
+        for (int i = 0; i < childNodes.getLength(); i++) {
+            Node node = childNodes.item(i);
+            categoryImages.add(getMediaFromPage(context, node));
+        }
+
+        return categoryImages;
+    }
+
+    private static Media getMediaFromPage(Context context, Node node) {
+        Media media = new Media(null,
+                getImageUrl(node),
+                getFileName(node),
+                getDescription(node),
+                getDataLength(node),
+                getDateCreated(node),
+                getDateCreated(node),
+                getCreator(node)
+        );
+
+        media.setLicense(getLicense(node));
+
+        return media;
+    }
+
+    private static String getFileName(Node document) {
+        Element element = (Element) document;
+        return element.getAttribute("title");
+    }
+
+    private static String getDescription(Node document) {
+        return getMetaDataValue(document, "ImageDescription");
+    }
+
+    private static String getLicense(Node document) {
+        return getMetaDataValue(document, "License");
+    }
+
+    private static String getCreator(Node document) {
+        String artist = getMetaDataValue(document, "Artist");
+        if (artist != null) {
+            return Jsoup.parse(artist).text();
+        }
+        return null;
+    }
+
+    private static Date getDateCreated(Node document) {
+        String dateTime = getMetaDataValue(document, "DateTime");
+        Timber.d("Date time is %s", dateTime);
```

Comment:
Could you please make this verbose at least? It spams my entire logcat screen each time any page loads. :)

Reply:
Will remove it altogether. Was just for debugging purposes :D